rskj-core/src/main/java/co/rsk/net/SnapshotProcessor.java

```diff
@@ -142,7 +146,20 @@ public SnapshotProcessor(Blockchain blockchain,
 
         this.checkHistoricalHeaders = checkHistoricalHeaders;
         this.parallel = isParallelEnabled;
-        this.thread = new Thread(new SyncMessageHandler("SNAP/server", requestQueue, listener) {
+        this.requestQueue = new LinkedBlockingQueue<>() {
+            @Override
+            public boolean offer(@Nonnull SyncMessageHandler.Job job) {
```

Comment:
I like this, but I’m afraid this function might end up a bit hidden. Maybe it’s just a matter of preference, but I would do this override in a new class that extends LinkedBlockingQueue.